Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -188,21 +188,22 @@ CandidateExaminationResult ScrollAnchoringController::examineCandidate(Element&
     }
 
     if (auto renderer = element.renderer()) {
-        // TODO: we need to think about position: absolute
         // TODO: figure out how to get scrollable area for renderer to check if it is maintaining scroll anchor
-        if (renderer->style().overflowAnchor() == OverflowAnchor::None || renderer->isStickilyPositioned() || renderer->isFixedPositioned() || renderer->isPseudoElement() || renderer->isAnonymousBlock() || (renderer->isAbsolutelyPositioned() && absolutePositionedElementOutsideScroller(*renderer, m_owningScrollableArea)))
+        auto boxRect = renderer->absoluteBoundingBoxRect();
+        bool intersects = containingRect.intersects(boxRect);
+
+        if (renderer->style().overflowAnchor() == OverflowAnchor::None || renderer->isStickilyPositioned() || renderer->isFixedPositioned() || renderer->isPseudoElement() || renderer->isAnonymousBlock() || (renderer->isAbsolutelyPositioned() && absolutePositionedElementOutsideScroller(*renderer, m_owningScrollableArea)) || (!intersects && renderer->style().containsPaint()))
```

Comment:
I don't see any mention of `contain: paint` in the spec (https://www.w3.org/TR/css-scroll-anchoring-1/).

Reply:
I think this is just an optimization when you know it's not intersecting.

This line is getting really long. I suggest you write it using a lambda.

Reply:
Fixed.